MIP/mip-25/README.md

```diff
@@ -0,0 +1,244 @@
+        let proposal_id = proxy_controller.next_proposal_id;
+        let proposal = Proposal {
+            creator,
+            proposal_id,
+            execution_hash,
+            stake,
+            approved: false,
+        };
+
+        proxy_controller.next_proposal_id = proxy_controller.next_proposal_id + 1;
+
+        smart_table::add(&mut proxy_controller.proposals, proposal_id, proposal);
+
+        event::emit(ProposalCreated {
+            creator,
+            proposal_id,
+        });
+    }
+
+    /// Approve proposal
+    public entry fun approve(caller: &signer, proposal_id: u64) acquires ProxyController {
+        let proxy_controller = borrow_global_mut<ProxyController>(@aptos_framework);
+        assert!(signer::address_of(caller) == proxy_controller.controller, EINVALID_CONTROLLER);
+        smart_table::borrow_mut(&mut proxy_controller.proposals, proposal_id).approved = true;
+
+        event::emit(ProposalApproved {
+            proposal_id,
+        });
+    }
+
+    /// Reject proposal
+    public entry fun reject(caller: &signer, proposal_id: u64) acquires ProxyController {
+        let proxy_controller = borrow_global_mut<ProxyController>(@aptos_framework);
+        assert!(signer::address_of(caller) == proxy_controller.controller, EINVALID_CONTROLLER);
+
+        let Proposal {
+            creator,
+            proposal_id: _,
+            execution_hash: _,
+            stake,
+            approved: _ } = smart_table::remove(&mut proxy_controller.proposals, proposal_id);
+
+        coin::deposit(creator, stake);
+        event::emit(ProposalRejected {
+            proposal_id,
+        });
+    }
+
+    /// Delegate Aptos Framework's signing cap to proxy providing proposal id for approved proposal and calling from
+    /// the proposed script
+    public fun delegate_to_proxy(proposal_id: u64) : signer acquires ProxyController {
+        let proxy_controller = borrow_global_mut<ProxyController>(@aptos_framework);
+    
+        let Proposal {
+            creator,
+            proposal_id: _,
+            execution_hash,
+            stake,
+            approved } = smart_table::remove(&mut proxy_controller.proposals, proposal_id);
+
+        assert!(approved, ENOT_APPROVED);
+        assert!(transaction_context::get_script_hash() == execution_hash, EEXECUTION_HASH_INVALID);
+
+        coin::deposit(creator, stake);
+        create_signer_with_capability(&proxy_controller.aptos_framework_signer_cap)
+    }
+
+    #[view]
+    /// Return the controller of the proxy
+    public fun get_controller(): address acquires ProxyController {
+        borrow_global<ProxyController>(@aptos_framework).controller
+    }
+}
+```
+**Steps:**
+1. Create the Move script you want to execute.  An example can be found below.
+2. With the hash of the script call `proxy::propose(execution_hash)` to initiate the request to proxy the script providing a refundable stake.  On off chain mechanism would track the newly created `proposal_id`
```

Comment:
Should this proxy also lend to a time lock proxy as well?

Reply:
Good idea.  Presumably you are talking about an expiration time on the proposal?